### PR TITLE
Added pl-fc-entails section

### PR DESCRIPTION
**Pull request by ad71**
- Added explanation of PL-FC-Entails algorithm
- Briefly explained forward and backward chaining for propositional logic
- Briefly explained horn clauses and definite clauses
- Added examples of usage
- Reran notebook to serialize execution counts
- Added tests for `pl_fc_entails`
- Updated README.md
- Fixed wrong filename

### CODE REVIEW

File: logic.ipynb [hunk 1]
```diff
@@ -1438,55 +1438,520 @@
...
+    "### Forward and backward chaining\n",
+    "Introducing this section, we said we will look at two algorithms to check if a sentence is entailed by the `KB`, \n",
+    "but here's a third one. \n",
+    "The difference here is that our goal now is to determine if a knowledge base of definite clauses entails a single proposition symbol *q* - the query.\n",
+    "There is a catch however, the knowledge base can only contain **Horn clauses**.\n",
+    "<br>\n",
+    "#### Horn Clauses\n",
+    "Horn clauses can be defined as a *disjunction* of *literals* with **at most** one positive literal. \n",
+    "<br>\n",
+    "A horn clause with exactly one positive literal is called a *definite clause*.\n",
```
Comment (MrDupin) [inline logic.ipynb]: I think 'horn' should be capitalized. There are other cases below where the word is not capitalized, you may need to take care of them too.

File: logic.ipynb [hunk 2]
```diff
@@ -1438,55 +1438,520 @@
...
+   "source": [
+    "pl_resolution(wumpus_kb, ~P22), pl_resolution(wumpus_kb, P22)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Forward and backward chaining\n",
+    "Introducing this section, we said we will look at two algorithms to check if a sentence is entailed by the `KB`, \n",
```
Comment (MrDupin) [inline logic.ipynb]: I find the 'Introducing this section' a bit awkward. I suggest removing it. Something like 'Previously, we said ...' works best in my opinion.

**Comment by ad71** [comment]
Thanks for the review @MrDupin . I'll fix these asap.

**Comment by norvig** [comment]
Oops -- conflicts now because of the other PR for this file.

**Comment by ad71** [comment]
Fixed now @norvig

**Pull request merged**